libs/service-portal/settings/access-control/src/screens/GrantAccess/GrantAccess.tsx

```diff
@@ -116,97 +140,111 @@ function GrantAccess() {
-            <Text variant="h5">
+            <Text variant="h5" as="span">
               {formatMessage({
                 id: 'service.portal.settings.accessControl:grant-form-label',
                 defaultMessage: 'Sláðu inn upplýsingar aðgangshafa',
               })}
             </Text>
           </GridColumn>
-          <GridColumn paddingBottom={2} span={['12/12', '12/12', '6/12']}>
-            <InputController
-              control={control}
-              id="toNationalId"
-              defaultValue=""
-              rules={
-                {
-                  required: {
-                    value: true,
-                    message: formatMessage({
-                      id:
-                        'service.portal.settings.accessControl:grant-required-ssn',
-                      defaultMessage: 'Skylda er að fylla út kennitölu',
-                    }),
-                  },
-                  validate: {
-                    value: (value: number) => {
-                      if (!kennitala.isValid(value)) {
-                        return formatMessage({
-                          id:
-                            'service.portal.settings.accessControl:grant-invalid-ssn',
-                          defaultMessage: 'Kennitalan er ekki gild kennitala',
-                        })
-                      }
-                    },
-                  },
-                } as ValidationRules
-              }
-              type="tel"
-              format="######-####"
-              label={formatMessage({
-                id: 'global:nationalId',
-                defaultMessage: 'Kennitala',
-              })}
-              placeholder={formatMessage({
-                id: 'global:nationalId',
-                defaultMessage: 'Kennitala',
-              })}
-              onChange={(value) => {
-                requestDelegation(value)
-              }}
-            />
-          </GridColumn>
-          <GridColumn paddingBottom={2} span={['12/12', '12/12', '6/12']}>
-            <Controller
-              control={control}
-              name="name"
-              defaultValue=""
-              render={({ onChange, value, name }) => (
+          <GridColumn span={['12/12', '12/12', '8/12']} paddingBottom={4}>
+            <div className={styles.inputWrapper}>
+              {name && (
                 <Input
-                  name={name}
-                  readOnly={true}
+                  name="name"
+                  value={name}
                   label={formatMessage({
                     id:
-                      'service.portal.settings.accessControl:grant-label-user',
-                    defaultMessage: 'Aðfangshafi',
+                      'service.portal.settings.accessControl:grant-form-access-holder',
+                    defaultMessage: 'Aðgangshafi',
+                  })}
+                  disabled
+                />
+              )}
+              <Box display={name ? 'none' : 'block'}>
```

Comment:
Ah. I guess it's pretty tricky to implement this behavior with good a11y. I wonder if this implementation loses the focus position. Should create a follow up task for island-ui.

Reply:
It should also announce the change somehow, eg with [live regions](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions).